code/game/objects/structures/door_assembly.dm

```diff
@@ -165,143 +170,188 @@
+	else if(W.iswelder())
+		if(!(istext(glass) || glass == TRUE || !anchored))
+			to_chat(user, SPAN_WARNING("\The [src] isn't ready to be welded yet. It has to have glass installed to remove the glass, or it has to be unsecured to deconstruct."))
+			return
 		var/obj/item/weldingtool/WT = W
-		if (WT.remove_fuel(0, user))
+		if(WT.remove_fuel(0, user))
 			playsound(src.loc, 'sound/items/welder_pry.ogg', 50, 1)
 			if(istext(glass))
-				user.visible_message("[user] welds the [glass] plating off the airlock assembly.", "You start to weld the [glass] plating off the airlock assembly.")
-				if(do_after(user, 40/W.toolspeed))
-					if(!src || !WT.isOn()) return
-					to_chat(user, "<span class='notice'>You welded the [glass] plating off!</span>")
+				user.visible_message("<b>[user]</b> starts welding the [glass] plating off the airlock assembly.", SPAN_NOTICE("You start welding the [glass] plating off the airlock assembly."))
+				if(do_after(user, 40 / W.toolspeed))
+					if(!src || !WT.isOn())
+						return
+					to_chat(user, SPAN_NOTICE("You weld the [glass] plating off."))
 					var/M = text2path("/obj/item/stack/material/[glass]")
 					new M(src.loc, 2)
-					glass = 0
-			else if(glass == 1)
-				user.visible_message("[user] welds the glass panel out of the airlock assembly.", "You start to weld the glass panel out of the airlock assembly.")
-				if(do_after(user, 40/W.toolspeed))
-					if(!src || !WT.isOn()) return
-					to_chat(user, "<span class='notice'>You welded the glass panel out!</span>")
+					glass = FALSE
+			else if(glass == TRUE)
+				user.visible_message("<b>[user]</b> starts welding the glass panel out of the airlock assembly.", SPAN_NOTICE("You start welding the glass panel out of the airlock assembly."))
+				if(do_after(user, 40 / W.toolspeed))
+					if(!src || !WT.isOn())
+						return
+					to_chat(user, SPAN_NOTICE("You weld the glass panel out."))
 					new /obj/item/stack/material/glass/reinforced(src.loc)
-					glass = 0
+					glass = FALSE
 			else if(!anchored)
-				user.visible_message("[user] dissassembles the airlock assembly.", "You start to dissassemble the airlock assembly.")
-				if(do_after(user, 40/W.toolspeed))
-					if(!src || !WT.isOn()) return
-					to_chat(user, "<span class='notice'>You dissasembled the airlock assembly!</span>")
+				user.visible_message("<b>[user]</b> starts disassembling the airlock assembly.", SPAN_NOTICE("You start disassembling the airlock assembly."))
+				if(do_after(user, 40 / W.toolspeed))
+					if(!src || !WT.isOn())
+						return
+					to_chat(user, SPAN_NOTICE("You disassemble the airlock assembly."))
 					dismantle()
 		else
-			to_chat(user, "<span class='notice'>You need more welding fuel.</span>")
+			to_chat(user, SPAN_WARNING("You need more welding fuel."))
+			return
+
+	else if(W.iswrench())
+		if(state != STATE_UNWIRED)
+			to_chat(user, SPAN_WARNING("\The [src] can only be wrenched once it's in an unwired state."))
```

Comment:
Perhaps instead of talking about states it should say something like
```suggestion
			to_chat(user, SPAN_WARNING("You have to remove the wiring before unwrenching \the [src]."))
```